control-plane/api-gateway/binding/validation.go

```diff
@@ -204,11 +208,59 @@ func validateTLS(gateway gwv1beta1.Gateway, tls *gwv1beta1.GatewayTLSConfig, res
 	return nil, err
 }
 
+// Envoy will silently reject any keys that are less than 2048 bytes long
+// https://github.com/envoyproxy/envoy/blob/main/source/extensions/transport_sockets/tls/context_impl.cc#L238
+const MinKeyLength = 2048
+
 func validateCertificateData(secret corev1.Secret) error {
-	_, _, err := common.ParseCertificateData(secret)
+	_, privateKey, err := common.ParseCertificateData(secret)
 	if err != nil {
 		return errListenerInvalidCertificateRef_InvalidData
 	}
+
+	err = validateKeyLength(privateKey)
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
#nit

```suggestion
	return validateKeyLength(privateKey)
```

Reply:
can do, I usually like to have the error checking be explicit but this works for me